### PR TITLE
Bug/csi 3038 node pods in status creating due to bad secrets corelati…

### PR DESCRIPTION
**Pull request by zingero**
…on (#167)

* changed secrets problem handle

* get the node pods

* add watch on service account

* remove status update

* remove logger from updateStatus

* add logger info on status update

* get when the operator sync the node pods

* add 40 seconds wait after restart

* remove updating IBMBlockCSI status

* add ds_restarted_key variable

* add ds_restarted_key to node SyncFn

* fix wrong variables

* remove declared restarted variables

* remove node syncer logger

* add handle to controller ImagePullBackOff

* answer some PR comments

* change some functions names

* add daemonSet_restarted parameters to verify annotations function

* add areAllPodImagesSynced check in the controller

* updated GetAnnotations function

* remove logger update outside a if

* changed the controller restart handle

* fix more code review comments

* add a name for the serviceaccount return value

* add errors.IsNotFound to getting the controller pod

* add Ctrl+Alt+L

* add CSI-3071 comment

* change the controller restart handle

* cahnge the logger type in the functions

* use logr.Logger in functions

* remove instance from restartControllerPodfromStatefulSet function

### CODE REVIEW

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 1]
```diff
@@ -52,12 +52,17 @@ import (
 	"github.com/IBM/ibm-block-csi-operator/pkg/internal/ibmblockcsi"
 	kubeutil "github.com/IBM/ibm-block-csi-operator/pkg/util/kubernetes"
 	oversion "github.com/IBM/ibm-block-csi-operator/version"
+	"github.com/go-logr/logr"
```
Comment (oriyarde) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:55]: I think these changes are [already in develop](https://github.com/IBM/ibm-block-csi-operator/blob/3c6d21d87f618231937ae63bbd243ae694e0dbbc/pkg/controller/ibmblockcsi/ibmblockcsi_controller.go#L55)

Comment (zingero) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:55]: This PR is for 1.5.1, not develop.
I changed the base branch accordingly.

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**